src/test/team_scoreboard.test.ts

```diff
@@ -0,0 +1,258 @@
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[3]), 50);
+
+                assert.strictEqual(scoreboard.getTeam(FIRST_TEAM_NAME).getScore(), 30);
+                assert.strictEqual(scoreboard.getTeam(SECOND_TEAM_NAME).getScore(), 3);
+            });
+        });
+        describe("some users guess correctly multiple times", function () {
+            it("should increment each user's score", function () {
+                for (let i = 0; i < 20; i++) {
+                    scoreboard.updateScoreboard(USER_TAG, FOUR_USER_IDS[0], AVATAR_URL, 1, 50);
+                    if (i === 0) {
+                        scoreboard.updateScoreboard(USER_TAG, FOUR_USER_IDS[1], AVATAR_URL, 1, 50);
+                    }
+                    if (i % 10 === 0) {
+                        scoreboard.updateScoreboard(USER_TAG, FOUR_USER_IDS[2], AVATAR_URL, 1, 50);
+                    }
+                }
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[0]), 20);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[0]), 1000 * 1.1);
+
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[1]), 1);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[1]), 50 * 1.1);
+
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[2]), 2);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[2]), 100);
+
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[3]), 0);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[3]), 0);
+
+                assert.strictEqual(scoreboard.getTeam(FIRST_TEAM_NAME).getScore(), 21);
+                assert.strictEqual(scoreboard.getTeam(SECOND_TEAM_NAME).getScore(), 2);
+            });
+        });
+        describe("all users have not guessed yet", function () {
+            it("should not increment each user's score", function () {
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[0]), 0);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[0]), 0);
+
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[1]), 0);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[1]), 0);
+
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[2]), 0);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[2]), 0);
+
+                assert.strictEqual(scoreboard.getPlayerScore(FOUR_USER_IDS[3]), 0);
+                assert.strictEqual(scoreboard.getPlayerExpGain(FOUR_USER_IDS[3]), 0);
+
+                assert.strictEqual(scoreboard.getTeam(FIRST_TEAM_NAME).getScore(), 0);
+                assert.strictEqual(scoreboard.getTeam(SECOND_TEAM_NAME).getScore(), 0);
+            });
+        });
+    });
+});
+
+beforeEach(function () {
```

Comment:
Both of the `beforeEach()` in this file are in the same scope. Is this one redundant?

Reply:
Whoops, thanks! It is